src/components/NavigationStack/NavigationStack.tsx

```diff
@@ -0,0 +1,96 @@
+import {
+    useState,
+    createContext,
+    useMemo,
+    useCallback,
+    ReactElement,
+    forwardRef,
+    memo,
+    useImperativeHandle,
+    Children,
+    isValidElement,
+    FC,
+} from 'react';
+
+export type Props = {
+    initialRouteName?: string;
+    children: ReactElement | ReactElement[];
+};
+
+export type NavigationStackHandle = {
+    currentRoute: string;
+    push: (route: string) => void;
+    pop: () => void;
+    navigate: (route: string) => void;
+    goBack: () => void;
+};
+
+const NavigationStack = ({ initialRouteName, children: childrenProp }: Props, ref: any) => {
+    // NavigationStack.Item can only be the direct children
+    const children = useMemo(
```

Comment:
remove mapping over children, instead register Navigation.Item to the context. on unmount, it should deregister.

Reply:
removed